homework-submission/todo-app/src/App.js

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react';
+import { todoList } from './services/todoContents';
+import Todo from './components/todo';
+import './App.css';
+import Input from './components/input';
+import uuid from 'uuid';
+function App() {
+  const initialState = {
+    id: uuid.v4(),
+    description: '',
+    deadline: '',
+    done: false,
+  };
+
+  const [todoListState, setTodoListState] = useState(todoList);
+  const [descAndDate, setDescAndDate] = useState(initialState);
+  const [errors, setErrors] = useState({});
+
+  const handleDone = id => {
+    const todoIndex = todoListState.findIndex(todo => todo.id === id);
+    const newTodoListState = [...todoListState];
+    const todo = newTodoListState[todoIndex];
+    todo.done = !todo.done;
+    setTodoListState(newTodoListState);
+  };
+
+  const removeHandler = id => {
+    const todoList = todoListState.filter(todo => todo.id !== id);
+    setTodoListState(todoList);
+  };
+
+  const handleChange = ({ target: input }) => {
+    setDescAndDate({
+      ...descAndDate,
+      [input.name]: input.value,
+    });
+  };
+
+  const validate = () => {
+    if (descAndDate.description === '') {
+      setErrors({ message: 'Description is required!' });
+      return errors;
+    }
+    if (descAndDate.deadline === '') {
+      setErrors({ message: 'Deadline is required!' });
+      return errors;
+    }
+  };
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    const errors = validate();
+    if (errors) return;
+    setTodoListState([...todoListState, descAndDate]);
+    setDescAndDate(initialState);
+    setErrors({});
+  };
+
+  return (
+    <div className="App">
+      <ul>
+        {todoListState.map(todo => (
+          <Todo key={todo.id} {...todo} handleDone={handleDone} onDelete={removeHandler} />
+        ))}
+      </ul>
+      <form className="form" onSubmit={handleSubmit}>
+        {errors.message ? <div className="alert">{errors.message}</div> : null}
```

Comment:
Cool that you implemented validation and are showing the errors! I do notice that when I add a new todo the deadline input doesn't change, but when I want to add another todo it says that I have to define a deadline.